packages/fast-components-styles-msft/src/utilities/density.ts

```diff
@@ -52,15 +52,41 @@ export function heightNumber(lines: number = 1): DesignSystemResolver<number> {
  * @param designSystem The design system config.
  */
 export function getDensityCategory(designSystem: DesignSystem): DensityCategory {
+    const densityValue: number = getDesignSystemValue("density")(designSystem);
     const category: DensityCategory =
-        designSystem.density >= 2
+        densityValue >= 2
             ? DensityCategory.spacious
-            : designSystem.density <= -2
+            : densityValue <= -2
                 ? DensityCategory.compact
                 : DensityCategory.normal;
     return category;
 }
 
+/**
+ * Returns an offset based on the higher-level category for the density setting.
+ * Used to adjust things like type size and sizing that is based on the category rather than individual density.
+ *
+ * @param compactOffset The adjustment when the category is "compact"
+ * @param spaciousOffset The adjustment when the category is "spacious"
+ * @param normalOffset The adjustment when the category is "normal"
+ */
+export function getOffsetForDensityCategory(
```

Comment:
maybe `densityCategorySwitch`  - then this can accept any generic value and return that value depending on the density category